mne/io/nirx/tests/test_nirx.py

```diff
@@ -36,6 +37,18 @@ def test_nirx_hdr_load():
     assert raw.info['sfreq'] == 12.5
 
 
+@requires_testing_data
+def test_nirx_dat_warn():
+    """Test reading NIRX files when missing data."""
+    os.rename(fname_nirx_15_2_short + "/NIRS-2019-08-23_001.dat",
+              fname_nirx_15_2_short + "/NIRS-2019-08-23_001.tmp")
+    fname = fname_nirx_15_2_short + "/NIRS-2019-08-23_001.hdr"
+    with pytest.raises(RuntimeWarning, match='A single dat'):
+        read_raw_nirx(fname, preload=True)
+    os.rename(fname_nirx_15_2_short + "/NIRS-2019-08-23_001.tmp",
+              fname_nirx_15_2_short + "/NIRS-2019-08-23_001.dat")
```

Comment:
this is for me dangerous as if the lines above fail you will never revert the renaming.

Reply:
Agreed. Could I not have a test for this? It will drop the code coverage :(

Or is there another way you would suggest?

Reply:
@rob-luke the way we usually solve this is with the pattern:
```
def test_whatever(tmpdir):  # pytest builtin tmpdir fixture
    shutuil.copytree(orig_dir, str(tmpdir))
    # mess with files in tmpdir
    ...
    read_raw_nirx(tmpdir.join('whatever.ext'))
```
pytest takes care of creating and destroying the `tmpdir` directory

Reply:
beautiful. Thanks